### ISSUE TITLE
experiment_family not working properly

### ISSUE THREAD

**Issue by paolap** [coecms/clef#80]

**Comment by paolap** [comment]
saw562@raijin4 ~ $ clef --local cmip5 --model=ACCESS1.0 --experiment_family=RCP --frequency=mon --variable=tas --ensemble=r1i1p1
/g/data1/rr3/publications/CMIP5/output1/CSIRO-BOM/ACCESS1-0/1pctCO2/mon/atmos/Amon/r1i1p1/latest/tas
/g/data1/rr3/publications/CMIP5/output1/CSIRO-BOM/ACCESS1-0/abrupt4xCO2/mon/atmos/Amon/r1i1p1/latest/tas
/g/data1/rr3/publications/CMIP5/output1/CSIRO-BOM/ACCESS1-0/amip/mon/atmos/Amon/r1i1p1/latest/tas
/g/data1/rr3/publications/CMIP5/output1/CSIRO-BOM/ACCESS1-0/historical/mon/atmos/Amon/r1i1p1/latest/tas
/g/data1/rr3/publications/CMIP5/output1/CSIRO-BOM/ACCESS1-0/piControl/mon/atmos/Amon/r1i1p1/latest/tas
/g/data1/rr3/publications/CMIP5/output1/CSIRO-BOM/ACCESS1-0/rcp45/mon/atmos/Amon/r1i1p1/latest/tas
/g/data1/rr3/publications/CMIP5/output1/CSIRO-BOM/ACCESS1-0/rcp85/mon/atmos/Amon/r1i1p1/latest/tas

**Comment by paolap** [comment]
The reason for that is that we don't have experiment_family in the attributes.
To fix that we decided to change local search so it will add a filter(experiment like "...%") when experiment_family is present.

To make this work properly:
 - cli experiment_family can accept only one value (fixed in cmip5_args)
 - experiment_family in datasets_constraints has to be a list (fixed in cli.py around line 248)
 - as we did for var if experiment-family we add an extra filter to local_search in code.py
       .filter(C5Dataset.experiment.like(family_dict[family]))
where family_dict is {'RCP' : 'rcp%' , 'decadal': 'decadal%' ...}

it works but now local_search has all this options which modify the search and lots of repetition as a consequence, would be nice to be able to add only the relevant bit of query

**Comment by paolap** [comment]
I've changed the search tidied up the code to keep into consideration that some families have couldn't be captured by one expression to:
.filter(C5Dataset.experiment.like(any_(family_dict[family]))

I've also changed the code so it uses only one main search for both cmip5 and cmip6 and add one line filter for 'variable' and/or 'experiment_family' when defined in CMIP5.

also moved where I adjust the experiment_family from dataset_constraints to terms definition.
I've run tests and also try different queries and all seems to work.
In branch "localfix" which was previously merged with "tests"
